qa/rpc-tests/getblocktemplate.py

```diff
@@ -51,15 +61,15 @@ def run_test(self):
         # Test 5: General checks
         tmpl = node.getblocktemplate()
         assert_equal(16, len(tmpl['noncerange']))
+        # should be proposing height 2, since current tip is height 1
+        assert_equal(2, tmpl['height'])
 
         # Test 6: coinbasetxn checks
-        assert('foundersreward' in tmpl['coinbasetxn'])
         assert(tmpl['coinbasetxn']['required'])
 
-        # Test 7: hashFinalSaplingRoot checks
-        assert('finalsaplingroothash' in tmpl)
-        finalsaplingroothash = '3e49b5f954aa9d3545bc6c37744661eea48d7c34e3000d82b7f0010c30f4c2fb'
-        assert_equal(finalsaplingroothash, tmpl['finalsaplingroothash'])
+        # Test 7: blockcommitmentshash checks
+        assert('blockcommitmentshash' in tmpl)
+        assert('00' * 32 != tmpl['finalsaplingroothash'])
```

Comment:
This should be checking that `defaultroots` exists, and that `blockcommitmentshash` exists in that. It should also check the existence of the other expected fields, and ideally verify them. Non-blocking, we can improve this in a subsequent PR (as this is would effectively be #5223).